### DISCUSSION TITLE
Question about Pod API

### DISCUSSION THREAD

**Discussion by sfc-gh-aivanou** [fabric8io/kubernetes-client#4494] [Q&A] [answered]
Hello.
I am curious about some API decisions, in particular why:
```
kubernetesClient.pods().inNamespace("my_namespace").withName("my_pod")
```

returns `PodResource`, but

```
kubernetesClient.pods().inNamespace("my_namespace").list().getItems().get(0)
```
returns `Pod` object?

It feels like inconsistency in API.

**Comment by shawkins** [comment, marked as answer]
> It feels like inconsistency in API.

A PodResource, or any class extending Resource, is an entry point for further operations - delete, get, patch, etc. that operate over / return an actual instance of the Pod.

list is considered a similar operation - it returns the full response from the server, which includes the list metadata and actual instances.

Are you looking for something like the KubernetesClient.resourceList(...).resources() method that returns a list of Resources?

**Reply by sfc-gh-aivanou** [reply]
Got it, thank you very much for clarification!
